### ISSUE TITLE
After migration to 13.x gitlab no longer starts

### ISSUE THREAD

**Issue by tomastzn** [sameersbn/docker-gitlab#2236]
My usual process:
1. Follow instructions to create backup.
2. Upgrade to latest minor version in the next newer release. (from 12.9.2 to 12.10.6)
3. Repeat if more releases with greater major versions available. (from 12.10 to 13.0)
I was doing this because there were issues in the past if I skipped over releases.

The issue I'm reporting today is that I was able to get from 12.9.2 to 12.10.6-1, but then any attempt to go to 13.x results in this line being printed in the logs:
`/sbin/entrypoint.sh: line 19:   771 Terminated              /usr/bin/supervisord -nc /etc/supervisor/supervisord.conf`
The number `771` changes, not sure if that's a pid or not.

As to my environment, I'm using docker compose, using vanilla redis 5 and postgresql 11.

And can not go back because of this error:
```
gitlab2-gitlab | ERROR: 
gitlab2-gitlab |   Cannot downgrade from GitLab version 13.4.2 to 12.10.6.
gitlab2-gitlab |   Only upgrades are allowed. Please use sameersbn/gitlab:13.4.2 or higher.
gitlab2-gitlab |   Cannot continue. Aborting!
```
What do I do get back to 12.x that works? I've been on this for two days now, the frustration is setting in :(

**Comment by tomastzn** [comment]
I was able to recover back to 12.10.6-1 with the following steps:

1. Copy backup from NAS to the host where Gitlab in docker is running.
2. Remove all volumes for postgres and gitlab. They will be recreated on the next docker-compose command.
3. Start up postgres and then run the commands to setup the required extensions(using `docker-compose exec`). This effectively resulted in new, empy database for gitlab.
4. Run `docker-compose run --rm gitlab app:init`. This is supposed to initialize everything but run nothing.
5. Run `docker-compose run --rm gitlab app:rake db:setup` to setup the DB. Not sure if this was needed, but it looked like good idea.
6. Copy the backup TAR file back into the gitlab volume into folder `backups`.
7. Run `docker-compose run --rm gitlab app:rake gitlab:backup:restore`. This command offered only one backup to choose from, the one copied over in previous step.
8. Start gitlab normally.

The problem reported in this issue remains. I can not move to 13.x because the `supervisord` is terminating during startup. I'm willing to do some debugging if anyone has ideas.

**Comment by tomastzn** [comment]
Not sure what was the reason for this not working before, but it worked now. I suspect something was being left behind from previous versions and it was causing issue for the `supervisord` at startup.
With following process(almost the same as above), I was able to migrate from `12.10.6-1` to `13.4.2`:

1. Copy backup from NAS to the host where Gitlab in docker is running.
2. Remove all volumes for postgres and gitlab. They will be recreated on the next docker-compose command.
3. Start up postgres (I didn't run any commands to create the extensions, as that seems to be handled by next steps).
4. Run `docker-compose run --rm gitlab app:init`. This is supposed to initialize everything.
5. Run `docker-compose run --rm gitlab app:rake db:setup` to setup the DB. Not sure if this was needed, but it looked like good idea.
6. Copy the backup TAR file back into the gitlab's volume into folder `backups`.
**At this point, the docker compose file still points to `12.10.6-1` image of gitlab.** 
7. Run `docker-compose run --rm gitlab app:rake gitlab:backup:restore`. This command offered only one backup to choose from, the one copied over in previous step.
8. Modify the docker compose file **to point to version `13.4.2` of gitlab**.
9. Start gitlab normally with `docker-compose up -d` command.

In short, it seems the upgrade from 12.10 to 13.4 needed a clean start, followed by backup restore and then regular start.
I consider this issue resolved, but will keep it open in case there are any questions.

**Comment by QuickJack** [comment]
I upgraded my sameersbn/gitlab:12.9.5 to sameersbn/gitlab:12.10.6-1 and do have the some problem in the logs of my production server:

`/sbin/entrypoint.sh: line 19: 771 Terminated /usr/bin/supervisord -nc /etc/supervisor/supervisord.conf`

The server is running fine for about 3 weeks. I did not test 13.x yet. Are there any problems if I try to upgrade? What exactly is this supervisord thing doing? Why is GitLab working fine without supervisord running?

**Comment by QuickJack** [comment]
The relevant part of my log is as follows:

```
/sbin/entrypoint.sh: line 19:  1953 Terminated              /usr/bin/supervisord -nc /etc/supervisor/supervisord.conf
2020-11-02 11:28:55,865 CRIT Supervisor running as root (no user in config file)
2020-11-02 11:28:55,865 INFO Included extra file "/etc/supervisor/conf.d/cron.conf" during parsing
2020-11-02 11:28:55,865 INFO Included extra file "/etc/supervisor/conf.d/gitaly.conf" during parsing
2020-11-02 11:28:55,865 INFO Included extra file "/etc/supervisor/conf.d/gitlab-pages.conf" during parsing
2020-11-02 11:28:55,865 INFO Included extra file "/etc/supervisor/conf.d/gitlab-workhorse.conf" during parsing
2020-11-02 11:28:55,865 INFO Included extra file "/etc/supervisor/conf.d/groups.conf" during parsing
2020-11-02 11:28:55,865 INFO Included extra file "/etc/supervisor/conf.d/mail_room.conf" during parsing
2020-11-02 11:28:55,865 INFO Included extra file "/etc/supervisor/conf.d/nginx.conf" during parsing
2020-11-02 11:28:55,865 INFO Included extra file "/etc/supervisor/conf.d/puma.conf" during parsing
2020-11-02 11:28:55,865 INFO Included extra file "/etc/supervisor/conf.d/sidekiq.conf" during parsing
2020-11-02 11:28:55,865 INFO Included extra file "/etc/supervisor/conf.d/sshd.conf" during parsing
2020-11-02 11:28:55,869 INFO RPC interface 'supervisor' initialized
2020-11-02 11:28:55,869 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2020-11-02 11:28:55,869 INFO supervisord started with pid 1
```

**Comment by tomastzn** [comment]
Don't see anything left to discuss here, will close the issue.
